packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1473,6 +1485,144 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
   }
 
+  SelectionResult _updateSelectionEdgeByWord(Offset globalPosition, {required bool isEnd}) {
```

Comment:
I mentioned this before in my old review but just to double check, it's not possible to deduplicate this code with what we have in EditableText is it?  Assuming that EditableText also has similar code to extend selection by word.

Reply:
The code for extending selection by word is in `RenderEditable` here https://github.com/flutter/flutter/blob/de004293d83c4c7f9087b6382e4cb59c5d98b3f7/packages/flutter/lib/src/rendering/editable.dart#L2066-L2090 . I think the main difference is that we are not using the drag origin position to hold the origin word in place, but also a selection in `SelectionArea` can span multiple `Selectable`s while one `EditableText` contains the entire selection so the implementation details are a little different.